CMakeLists.txt

```diff
@@ -16,7 +16,12 @@ set(CMAKE_C_STANDARD 11)
 set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_C_EXTENSIONS OFF)
 
-set(CMAKE_CXX_STANDARD 17)
+set(MINIMUM_REQUIRED_CXX_STANDARD 17)
+
+if((NOT CMAKE_CXX_STANDARD) OR (CMAKE_CXX_STANDARD LESS MINIMUM_REQUIRED_CXX_STANDARD))
+    set(CMAKE_CXX_STANDARD ${MINIMUM_REQUIRED_CXX_STANDARD})
+endif()
```

Comment:
Question:
Are we sure that our old code never worked with C++14?
If so, we probably want to continue supporting it too.